### DISCUSSION TITLE
A Discussion on Dates

### DISCUSSION THREAD

**Discussion by technopahadi** [apache/airflow#27291] [Ideas]
The way some of the dates are portrayed in the UI and used in the code (like in Timetables) seems a bit confusing. It could possibly be a bit more simpler? Let me explain.

### Confusion 1

In the Grid View, the top right of the page has a `Next Run` label that shows a date. This is the `logical date` of the next scheduled run, or the start of the next DAG Run's data interval. When hovered, it shows a tooltip with `Run After` which denotes the actual "execution time" of the upcoming run.

### Possible Improvement

I think if the label simply showed the execution time of the next run instead, i.e., the `Next Run` actually depicted the value from `Run After`, that'd be more intuitive. The label itself (`Next Run`) can be renamed to something else (like `Next Scheduled Execution`) if it conflicts with some internal terminologies.

---

### Confusion 2

The `Run` label next to the DAG ID (inside the `#react-container`) depicts the:
1. Logical date for a scheduled run
2. Logical date of the last scheduled run for a manual run
3. Execution time for a manual run where `schedule=None`.

### Possible Improvement

If it always represents the execution time like `(3)`, then that'd be more intuitive imho.

---

### Confusion 3

`Run Id` in the DAG Run details depicts:
1. The logical date (start of data interval) for a scheduled run
2. Execution time (or maybe dagrun object creation time) for a manual run.

### Possible Improvement

The behaviour for manual run makes sense, but if the same could happen for scheduled run as well, that'd probably make more sense. With Timetables, [this is customizable](https://airflow.apache.org/docs/apache-airflow/stable/howto/timetable.html#changing-generated-run-id) since 2.4. Maybe it can be made the default behaviour instead?

**Comment by potiuk** [comment]
I think if you attempt to make a PR and show it before/and after on examples - then the discussion might be better (and having a PR you will see how difficult/easy it is).

BTW. I have no opinion on the subject this is just a comment that might move the discussion in the right direction
